src/test/java/games/strategy/engine/framework/ArgParserTest.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.engine.framework;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+
+import org.junit.Test;
+
+public class ArgParserTest {
+
+
+  @Test
+  public void argsTurnIntoSystemProps() {
+    assertThat("check precondition, system property for our test key should not be set yet.",
+        System.getProperty(TestData.propKey), nullValue());
+
+    boolean result = ArgParser.handleCommandLineArgs(
+        TestData.sampleArgInput, TestData.samplePropertyNameSet);
+
+    assertThat("prop key was supplied as an available value, "
+            + " which was passed as a test value - everything should "
+            + "have parsed well, expect true result",
+        result, is(true));
+    assertThat("system property should now be set to our test value",
+        System.getProperty(TestData.propKey), is(TestData.propValue));
```

Comment:
One concern I have in this fixture is that we are leaking state across test cases.  Specifically, `ArgParser` is setting system properties that are not being cleared between tests.  I didn't scrutinize each test case to determine if any of them are affected by this leakage (this would be somewhat painful as the order in which the tests are run is not necessarily deterministic, and there are a lot of possible permutations here).

The brute force solution would be to just clear all system properties for keys known to be created in the fixture in an `@After`.

This might be overkill, but you could also introduce something like a `SystemProperties` interface that is used to abstract access to `System#getProperty()` and `System#setProperty()`.  Then add a package-private overload for `handleCommandLineArgs()` that takes an instance of this interface.  The original `handleCommandLineArgs()` would just call the package-private version with a default `SystemProperties` implementation that delegates to `java.lang.System`.